lms/djangoapps/courseware/tabs.py

```diff
@@ -3,44 +3,347 @@
 perform some LMS-specific tab display gymnastics for the Entrance Exams feature
 """
 from django.conf import settings
-from django.test.client import RequestFactory
 from django.utils.translation import ugettext as _
 
 from courseware.access import has_access
 from courseware.entrance_exams import user_must_complete_entrance_exam
-from student.models import CourseEnrollment, EntranceExamConfiguration
-from xmodule.tabs import CourseTabList
+from openedx.core.djangoapps.course_views.course_views import CourseViewTypeManager, CourseViewType, StaticTab
+from student.models import CourseEnrollment
+from xmodule.tabs import CourseTab, CourseTabList, key_checker
 
-from util import milestones_helpers
+
+class EnrolledCourseViewType(CourseViewType):
+    """
+    A base class for any view types that require a user to be enrolled.
+    """
+    @classmethod
+    def is_enabled(cls, course, user=None):
+        if not user:
```

Comment:
Do you mean `if user is None`? User should either be an `User` or `None`, right?